src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerContext.cs

```diff
@@ -51,43 +49,31 @@ internal set
         /// Indicates whether pre-generated serialization logic for types in the context
         /// is compatible with the run time specified <see cref="JsonSerializerOptions"/>.
         /// </summary>
-        internal bool CanUseSerializationLogic
+        internal bool CanUseFastPathSerializationLogic(JsonSerializerOptions options)
```

Comment:
Moves this flag from `JsonSerializerContext` to `JsonSerializerOptions`. This accounts for the fact that, since .NET 7, a `JsonSerializerContext` can be referenced by multiple `JsonSerializerOptions` instances, as such the invalidation logic needs to be evaluated independently for the configuration of every `JsonSerializerOptions` instance.

While not strictly related to this change, I added this in order to simplify the source gen testing infrastructure (removing the requirement for a `options => MyContext(options)` delegate being passed on every test suite declaration.

Reply:
nit: if it's not too disruptive can you please undo the test changes that don't concern this feature and make them in a follow up PR?

Reply:
I should have clarified in my earlier comment that this is part of a wider refactoring of the "JsonSerializerWrapper" source gen implementations, which were necessary to accommodate some of the tests in the existing required members test suite.

TL;DR it involves implementing a few missing methods and moving from the JsonSerializerContext overloads to the JsonSerializerOptions. These are functionally equivalent since the removal of reflection fallback in .NET 7. The diff of the JsonSerializerWrapper should clarify what exactly got updated.

So to answer your original question, even though I could remove them, they would necessitate refactoring of the test suite itself.

Reply:
No I get that. It just seemed noisy when I first started reading but hey I got through it.